staging/src/k8s.io/client-go/tools/cache/shared_informer_test.go

```diff
@@ -18,34 +18,59 @@ package cache
 
 import (
 	"fmt"
+	"math"
 	"strings"
 	"sync"
 	"testing"
 	"time"
 
 	"k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/clock"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/wait"
 	fcache "k8s.io/client-go/tools/cache/testing"
 )
 
+const timeFmt = "15:04:05.999999999"
+
 type testListener struct {
-	lock              sync.RWMutex
-	resyncPeriod      time.Duration
-	expectedItemNames sets.String
-	receivedItemNames []string
-	name              string
+	t                *testing.T
+	name             string
+	clock            clock.PassiveClock
+	resyncPeriod     time.Duration
+	checkTiming      bool // whether handle() checks timing
```

Comment:
needs better godoc -- afaict, this ensure that handle checks to make sure that objects are not updated *faster* than the resync period (as opposed to checkRecency, which confirms that objects are not updated too slowly)